src/tests/unit/tests/background/target-page-controller.test.ts

```diff
@@ -2,748 +2,330 @@
-            )
-            .returns(() => tabContext)
-            .verifiable(Times.once());
-    }
+    let mockTabContextFactory: IMock<TabContextFactory>;
+    let mockBrowserAdapter: SimulatedBrowserAdapter;
+    let mockDetailsViewController: SimulatedDetailsViewController;
+    let mockTabInterpreters: DictionaryNumberTo<IMock<Interpreter>>;
 
-    function createTabControllerWithoutFeatureFlag(tabContextMap: TabToContextMap): TargetPageController {
-        return new TargetPageController(
-            tabContextMap,
-            mockBroadcasterStrictMock.object,
-            mockChromeAdapter.object,
-            mockDetailsViewController.object,
-            tabContextFactoryMock.object,
-            loggerMock.object,
-        );
-    }
+    const EXISTING_WINDOW_ID = 101;
+    const EXISTING_WINDOW = { id: EXISTING_WINDOW_ID } as chrome.windows.Window;
+
+    const EXISTING_ACTIVE_TAB_ID = 1;
+    const EXISTING_ACTIVE_TAB = { id: EXISTING_ACTIVE_TAB_ID, windowId: EXISTING_WINDOW_ID, active: true } as chrome.tabs.Tab;
+
+    const EXISTING_INACTIVE_TAB_ID = 2;
+    const EXISTING_INACTIVE_TAB = { id: EXISTING_INACTIVE_TAB_ID, windowId: EXISTING_WINDOW_ID, active: false } as chrome.tabs.Tab;
+
+    const NEW_TAB_ID = 3;
+    const NEW_TAB = { id: NEW_TAB_ID, windowId: EXISTING_WINDOW_ID, active: true } as chrome.tabs.Tab;
 
     beforeEach(() => {
-        loggerMock = Mock.ofType<Logger>();
+        mockLogger = Mock.ofType<Logger>();
         mockBroadcasterStrictMock = Mock.ofType<TabContextBroadcaster>(undefined, MockBehavior.Strict);
-        mockChromeAdapter = Mock.ofType<BrowserAdapter>();
-        mockDetailsViewController = Mock.ofType<DetailsViewController>();
-        featureFlagStoreMock = Mock.ofType(FeatureFlagStore);
-        telemetryEventHandlerMock = Mock.ofType(TelemetryEventHandler);
-
-        tabInterpreterMap = {};
-        tabContextFactoryMock = Mock.ofType(TabContextFactory);
-        mockDetailsViewController
-            .setup(dvc => dvc.setupDetailsViewTabRemovedHandler(It.isAny()))
-            .callback(cb => {
-                onDetailsViewTabRemoved = cb;
-            })
-            .verifiable(Times.once());
+        mockBroadcasterStrictMock.setup(m => m.getBroadcastMessageDelegate(It.isAny())).returns(_ => stubBroadcastDelegate);
+        mockBrowserAdapter = createSimulatedBrowserAdapter([EXISTING_ACTIVE_TAB, EXISTING_INACTIVE_TAB], [EXISTING_WINDOW]);
+        mockDetailsViewController = setupMockDetailsViewController();
+        mockTabInterpreters = {};
+        mockTabInterpreters[EXISTING_ACTIVE_TAB_ID] = Mock.ofType<Interpreter>();
+        mockTabInterpreters[EXISTING_INACTIVE_TAB_ID] = Mock.ofType<Interpreter>();
+        mockTabInterpreters[NEW_TAB_ID] = Mock.ofType<Interpreter>();
+        mockTabContextFactory = setupMockTabContextFactory(mockTabInterpreters);
 
         testSubject = new TargetPageController(
-            tabInterpreterMap,
+            {},
```

Comment:
should this be `mockTabInterpreters` ???

Reply:
No (the intention is to track those separately to be able to test this class's ability to do its own tracking/mutation of the tabToContextMap), but I didn't notice the first time around that the mutations to the tabToContextMap are externally observable contracts of this class's behavior, so you're right that we should be passing a value for this and testing those mutations. Updated tests.